### DISCUSSION TITLE
Reason for Overriden hashCode method in various classes?

### DISCUSSION THREAD

**Discussion by David10512799** [digitalpetri/ethernet-ip#71] [General]
Hello, 

I am curious about what the purpose of the Overridden hashCode methods are in a lot of the classes.  e.g. SendRRData.java

What are these used for and why?  Why were they built with the method that seems similar across all of them?

**Comment by kevinherron** [comment]
They're meant to be simple "data" classes that can be compared for equality. It's standard practice to implement `hashCode` when implementing `equals`.
